stylus-verifier/stylus-verifier-logic/src/docker.rs

```diff
@@ -74,65 +109,193 @@ async fn create_image(
         cargo_stylus_version,
         toolchain
     );
-    let mut child = Command::new("docker")
-        .arg("build")
-        .arg("-t")
-        .arg(name)
-        .arg(".")
-        .arg("-f-")
-        .stdin(std::process::Stdio::piped())
-        .spawn()
-        .context("failed to execute Docker build command")?;
-
-    let mut dockerfile = Vec::<u8>::new();
-    write!(
-            dockerfile,
-            "\
-            FROM --platform=linux/amd64 offchainlabs/cargo-stylus-base:{cargo_stylus_version} as base
-            RUN rustup toolchain install {toolchain}-x86_64-unknown-linux-gnu
-            RUN rustup default {toolchain}-x86_64-unknown-linux-gnu
-            RUN rustup target add wasm32-unknown-unknown
-            RUN rustup component add rust-src --toolchain {toolchain}-x86_64-unknown-linux-gnu
-            ",
-        ).expect("write into the vector should not fail");
-
-    child
-        .stdin
-        .as_mut()
-        .unwrap()
-        .write_all(&dockerfile)
-        .await
-        .context("failed to write dockerfile content into docker process stdin")?;
-    child.wait().await.context("wait failed")?;
+
+    let dockerfile = format!(
+        "FROM offchainlabs/cargo-stylus-base:{cargo_stylus_version} as base
+RUN rustup toolchain install {toolchain}-x86_64-unknown-linux-gnu
+RUN rustup default {toolchain}-x86_64-unknown-linux-gnu
+RUN rustup target add wasm32-unknown-unknown
+RUN rustup component add rust-src --toolchain {toolchain}-x86_64-unknown-linux-gnu
+"
+    );
+
+    let content = build_tar_with_dockerfile(&dockerfile)?;
+    let build_image_options = BuildImageOptions {
+        t: image_name.to_string(),
+        dockerfile: "Dockerfile".to_string(),
+        version: BuilderVersion::BuilderV1,
+        networkmode: "host".to_string(),
+        pull: true,
+        rm: true,
+        forcerm: true,
+        platform: "linux/amd64".to_string(),
+        ..Default::default()
+    };
+
+    let mut image_build_stream =
+        docker.build_image(build_image_options, None, Some(content.into()));
+
+    let mut output = vec![];
+    while let Some(result) = image_build_stream.next().await {
+        match result {
+            Ok(info) => {
+                if let Some(value) = info.stream {
+                    output.push(value)
```

Comment:
not sure what's returned here, but if it's some progress bar, then ig it makes sense to print it in logs?

Reply:
Added tracing logs in [c5ce76d](https://github.com/blockscout/blockscout-rs/pull/1074/commits/c5ce76da327956834ca5426ac3c8dae0d13686ef)